apps/web/lib/blog/services/post-service.ts

```diff
@@ -0,0 +1,156 @@
+import { serialize } from "next-mdx-remote/serialize";
+import remarkGfm from "remark-gfm";
+import { PostRepository } from "../repository/post-repository";
+import type {
+  BlogFilter,
+  BlogPost,
+  BlogPostListItem,
+  BlogSortOptions,
+} from "../types";
+import { formatDate, validateISODate } from "../utils/date";
+import { extractExcerpt } from "../utils/excerpt";
+import { calculateReadingTime } from "../utils/reading-time";
+
+export class PostService {
+  private repository: PostRepository;
+
+  constructor() {
+    this.repository = new PostRepository();
+  }
+
+  async getPost(slug: string): Promise<BlogPost | null> {
+    const postData = this.repository.getPostContent(slug);
+    if (!postData) return null;
+
+    const { content, data } = postData;
+
+    if (!validateISODate(data.date)) {
+      throw new Error(`Invalid date "${data.date}" in post ${slug}`);
+    }
+
+    const mdxSource = await serialize(content, {
+      mdxOptions: {
+        remarkPlugins: [remarkGfm],
+        rehypePlugins: [],
+      },
+    });
+
+    return {
+      id: slug,
+      slug,
+      title: data.title,
+      excerpt: data.excerpt || extractExcerpt(content),
+      category: data.category,
+      date: formatDate(data.date),
+      dateISO: data.date,
+      formattedDate: formatDate(data.date),
+      author: data.author,
+      featuredImage: data.featuredImage,
+      featured: data.featured,
+      readingTime: data.readingTime || calculateReadingTime(content),
```

Comment:
how often do we actually calculate the reading time? this seems expensive to be running on every pageload for a post